File: alpha-terminal-ai-server/app/domains/notification/domain/entity/notification.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Optional


@dataclass
class Notification:
    user_id: int
    title: str
    body: str
    is_read: bool = False
    link: Optional[str] = None
    id: Optional[int] = None
    created_at: datetime = field(default_factory=datetime.now)

File: alpha-terminal-frontend/features/notification/domain/model/notification.ts
export interface Notification {
    id: string
    title: string
    body: string
    type?: "info" | "warning" | "success" | "error"
    link?: string
    isRead: boolean
    createdAt: string
}

export interface NotificationListState {
    notifications: Notification[]
    unreadCount: number
    isLoading: boolean
    error: string | null
}
